### PR TITLE
truncateWords function should count words instead of characters

### PR DESCRIPTION
**Pull request by josephlacey**
Fix for #889

### CODE REVIEW

**Review by siepra — COMMENTED** [review]
What if I send a message like `thisisprobablytoolongmessagebut I'm gonnasenditanywaybecause I wanttotryifitwilldisplayproperlyand I'm messingwiththevalidator`? It won't get truncated as it's only 7 words long

**Comment by josephlacey** [comment]
> I think what we additionally need here is to limit the maximum number of characters to display. Otherwise someone can send a very long message without whitespaces and it won't ever get truncated

@siepra  Very nice catch.  I'll add this secondary validation.  50 characters?  @holmesworcester Thoughts on this?  The design doesn't account for this use case.

**Comment by holmesworcester** [comment]
Really it should be based on what fits and not characters but characters is okay too. If characters it should be a bit on the high side so it almost never cuts prematurely but sometimes runs over.

Also this is out of scope of this issue so quick fixes only!

**Comment by josephlacey** [comment]
> Really it should be based on what fits and not characters but characters is okay too. If characters it should be a bit on the high side so it almost never cuts prematurely but sometimes runs over.

@siepra @holmesworcester I've add a secondary check that shortens the message to 100 characters if the preview has less than 11 words but it still very long.  On my device that wraps to three lines.  The 11-word truncated version wraps to two lines, which is probably closer to around 60 characters.  I'm not sure if it's possible to check the screen width of the device and then determine the length of the message preview, but that seems beyond the scope of this issue.

Screenshot_20230402-134651_Debug Quiet

**Comment by holmesworcester** [comment]
I think this is a totally great solution!

File: packages/mobile/src/utils/functions/truncateWords/truncateWords.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 export const truncateWords = (str: string, num: number): string => {
   if (str.split(' ').length > num) {
     return str.split(' ').splice(0, num).join(' ') + '...'
+  } else if (str.length > 100) {
```
[the remark this answers is not in the document]
Comment (josephlacey) [reply packages/mobile/src/utils/functions/truncateWords/truncateWords.ts]: @siepra Very good idea.  Done.  I've also updated the function in ChannelTile and added a test for the new secondary validation.

**Review by siepra — APPROVED** [review]

**Pull request merged**